cvs diff: Diffing offapi
cvs diff: Diffing offapi/com
cvs diff: Diffing offapi/com/sun
cvs diff: Diffing offapi/com/sun/star
cvs diff: Diffing offapi/com/sun/star/accessibility
cvs diff: Diffing offapi/com/sun/star/animations
cvs diff: Diffing offapi/com/sun/star/auth
cvs diff: Diffing offapi/com/sun/star/awt
Index: offapi/com/sun/star/awt/MaxChildrenException.idl
===================================================================
RCS file: offapi/com/sun/star/awt/MaxChildrenException.idl
diff -N offapi/com/sun/star/awt/MaxChildrenException.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/MaxChildrenException.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,27 @@
+#ifndef __com_sun_star_awt_MaxChildrenException_idl__ 
+#define __com_sun_star_awt_MaxChildrenException_idl__ 
+
+#ifndef __com_sun_star_uno_Exception_idl__
+#include <com/sun/star/uno/Exception.idl>
+#endif
+
+//============================================================================= 
+
+module com {  module sun {  module star {  module awt {
+
+//============================================================================= 
+ 
+/** This exception is thrown when adding a child to a container that is full.
+
+	@see     XLayoutContainer
+	@see     com::sun::star::awt::XLayoutContainer::addChild
+ */
+published exception MaxChildrenException: com::sun::star::uno::Exception
+{ 
+}; 
+ 
+//============================================================================= 
+ 
+}; }; }; };
+
+#endif
Index: offapi/com/sun/star/awt/XDialog2.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XDialog2.idl
diff -N offapi/com/sun/star/awt/XDialog2.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XDialog2.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,26 @@
+#ifndef __com_sun_star_awt_XDialog2_idl__ 
+#define __com_sun_star_awt_XDialog2_idl__ 
+ 
+#ifndef __com_sun_star_awt_XDialog_idl__
+#include <com/sun/star/awt/XDialog.idl> 
+#endif 
+ 
+ 
+//============================================================================= 
+ 
+module com {  module sun {  module star {  module awt {
+ 
+//============================================================================= 
+ 
+/** Makes it possible to end a dialog.
+ */
+published interface XDialog2: com::sun::star::awt::XDialog
+{ 
+    void endDialog ( [in] long Result );
+}; 
+ 
+//============================================================================= 
+ 
+}; }; }; };  
+ 
+#endif 
Index: offapi/com/sun/star/awt/XLayoutContainer.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XLayoutContainer.idl
diff -N offapi/com/sun/star/awt/XLayoutContainer.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XLayoutContainer.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,75 @@
+#ifndef __com_sun_star_awt_XLayoutContainer_idl__ 
+#define __com_sun_star_awt_XLayoutContainer_idl__ 
+
+#ifndef __com_sun_star_awt_XLayoutConstrains_idl__ 
+#include <com/sun/star/awt/XLayoutConstrains.idl> 
+#endif 
+#ifndef __com_sun_star_awt_Rectangle_idl__ 
+#include <com/sun/star/awt/Rectangle.idl> 
+#endif 
+#ifndef __com_sun_star_beans_XPropertySet_idl__ 
+#include <com/sun/star/beans/XPropertySet.idl> 
+#endif 
+#ifndef __com_sun_star_container_XChild_idl__ 
+#include <com/sun/star/container/XChild.idl> 
+#endif
+#ifndef __com_sun_star_awt_MaxChildrenException_idl__ 
+#include <com/sun/star/awt/MaxChildrenException.idl> 
+#endif
+#ifndef __com_sun_star_awt_XLayoutUnit_idl__ 
+#include <com/sun/star/awt/XLayoutUnit.idl> 
+#endif
+
+//============================================================================= 
+
+module com {  module sun {  module star {  module awt {
+
+//============================================================================= 
+
+published interface XLayoutUnit;
+
+/** specifies the layout constraints for a surrounding container.
+ */
+published interface XLayoutContainer: com::sun::star::container::XChild
+{
+	/* Add/remove a child. Some containers provide an interface for
+	   adding children which you should use. */
+    void addChild( [in] com::sun::star::awt::XLayoutConstrains Child )
+		raises( com::sun::star::awt::MaxChildrenException ); 
+    void removeChild( [in] com::sun::star::awt::XLayoutConstrains Child );
+	sequence< com::sun::star::awt::XLayoutConstrains > getChildren();
+
+	/* Read/modify some child layout properties. XPropertySet provides a
+	   rich API that allows, e.g., for introspection.
+	   Should rarely be a need to use; containers shall provide an interface
+	   for setting properties more confortably. */
+    com::sun::star::beans::XPropertySet getChildProperties(
+		[in] com::sun::star::awt::XLayoutConstrains Child );
+
+	/* Set at initilization, this object should be notified when a containers state
+	   changes, that may affect its size, to queue a layout re-calculation. */
+	void setLayoutUnit( [in] com::sun::star::awt::XLayoutUnit Unit );
+
+	// TEMP: no need for this...
+	com::sun::star::awt::XLayoutUnit getLayoutUnit();
+
+	/* To be used by the parent for the purpose of layout management. 
+	   For widgets, use XWindow::setPosSize() */
+    void allocateArea( [in] com::sun::star::awt::Rectangle NewArea );
+
+	/* Used by the layouting unit to evaluate size damage, and force a
+	   re-allocation. */
+	com::sun::star::awt::Size getRequestedSize();
+	com::sun::star::awt::Rectangle getAllocatedArea();
+
+	/* For flow layouting (e.g. wrap-able labels). TODO: We might want to
+	   re-design this approach -- check XLayoutConstrains::calcAdjustedSize(). */
+	long getHeightForWidth( [in] long Width );
+	boolean hasHeightForWidth();
+};
+ 
+//============================================================================= 
+
+}; }; }; };
+
+#endif
Index: offapi/com/sun/star/awt/XLayoutFlow.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XLayoutFlow.idl
diff -N offapi/com/sun/star/awt/XLayoutFlow.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XLayoutFlow.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,29 @@
+#ifndef __com_sun_star_awt_XLayoutFlow_idl__ 
+#define __com_sun_star_awt_XLayoutFlow_idl__ 
+
+//============================================================================= 
+
+#include <com/sun/star/uno/XInterface.idl> 
+ 
+module com {  module sun {  module star {  module awt {
+ 
+//============================================================================= 
+ 
+/** Enables height-for-width layout negociations, which allows for label wrapping
+ *  and flow containers. Can be implemented by either a container or an ordinary widget;
+ *  whether its parent will honor it is another story, so keep implementing
+ *  getMinimumSize().
+ */
+published interface XLayoutFlow
+{
+	long getHeightForWidth( [in] long Width );
+
+	/* Allow the container/widget to toggle the functionality. */
+	boolean hasHeightForWidth();
+};
+ 
+//============================================================================= 
+ 
+}; }; }; };
+ 
+#endif 
Index: offapi/com/sun/star/awt/XLayoutFlowContainer.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XLayoutFlowContainer.idl
diff -N offapi/com/sun/star/awt/XLayoutFlowContainer.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XLayoutFlowContainer.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,25 @@
+#ifndef __com_sun_star_awt_XLayoutUnit_idl__ 
+#define __com_sun_star_awt_XLayoutUnit_idl__ 
+
+#include <com/sun/star/awt/XLayoutContainer.idl> 
+ 
+//============================================================================= 
+ 
+module com {  module sun {  module star {  module awt {
+ 
+//============================================================================= 
+ 
+/** Responsible to evaluate size damages and force a re-calculation. Containers
+    should let it know of state changes that may affects their size.
+ *  All children of a top-level window should share the same object.
+ */
+published interface XLayoutContainer
+{
+	void queueResize( [in] com::sun::star::awt::XLayoutContainer Container );
+};
+ 
+//============================================================================= 
+ 
+}; }; }; };
+ 
+#endif 
Index: offapi/com/sun/star/awt/XLayoutRoot.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XLayoutRoot.idl
diff -N offapi/com/sun/star/awt/XLayoutRoot.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XLayoutRoot.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,31 @@
+#ifndef __com_sun_star_awt_XLayoutRoot_idl__ 
+#define __com_sun_star_awt_XLayoutRoot_idl__ 
+ 
+#ifndef __com_sun_star_awt_XLayoutContainer_idl__ 
+#include <com/sun/star/awt/XLayoutContainer.idl> 
+#endif 
+#ifndef __com_sun_star_container_XNameAccess_idl__ 
+#include <com/sun/star/container/XNameAccess.idl> 
+#endif 
+#ifndef __com_sun_star_container_XNameContainer_idl__ 
+#include <com/sun/star/container/XNameContainer.idl> 
+#endif 
+ 
+//============================================================================= 
+ 
+module com {  module sun {  module star {  module awt {
+ 
+//============================================================================= 
+ 
+/** specifies an interface for a top-level layoutable widget
+ */
+published interface XLayoutRoot: com::sun::star::container::XNameAccess
+{
+    com::sun::star::awt::XLayoutContainer getLayoutContainer();
+}; 
+ 
+//============================================================================= 
+ 
+}; }; }; };  
+ 
+#endif 
Index: offapi/com/sun/star/awt/XLayoutUnit.idl
===================================================================
RCS file: offapi/com/sun/star/awt/XLayoutUnit.idl
diff -N offapi/com/sun/star/awt/XLayoutUnit.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/awt/XLayoutUnit.idl	8 Feb 2008 18:46:21 -0000	1.1.2.1
@@ -0,0 +1,29 @@
+#ifndef __com_sun_star_awt_XLayoutUnit_idl__ 
+#define __com_sun_star_awt_XLayoutUnit_idl__ 
+
+#ifndef __com_sun_star_awt_XLayoutContainer_idl__ 
+#include <com/sun/star/awt/XLayoutContainer.idl> 
+#endif
+
+//============================================================================= 
+ 
+module com {  module sun {  module star {  module awt {
+ 
+//============================================================================= 
+
+published interface XLayoutContainer;
+
+/** Responsible to evaluate size damages and force a re-calculation. Containers
+    should let it know of state changes that may affects their size.
+ *  All children of a top-level window should share the same object.
+ */
+published interface XLayoutUnit
+{
+	void queueResize( [in] com::sun::star::awt::XLayoutContainer Container );
+};
+ 
+//============================================================================= 
+ 
+}; }; }; };
+ 
+#endif
Index: offapi/com/sun/star/awt/makefile.mk
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/makefile.mk,v
retrieving revision 1.45
retrieving revision 1.45.16.1
diff -p -u -u -p -b -w -B -r1.45 -r1.45.16.1
--- offapi/com/sun/star/awt/makefile.mk	29 Jan 2008 15:11:34 -0000	1.45
+++ offapi/com/sun/star/awt/makefile.mk	11 Feb 2008 15:14:37 -0000	1.45.16.1
@@ -321,6 +321,17 @@ IDLFILES=\
 	XWindowPeer.idl\
 	XMenuExtended.idl
 
+.IF "$(ENABLE_LAYOUT)" == "TRUE"
+IDLFILES+=\
+	MaxChildrenException.idl\
+	XDialog2.idl\
+	XLayoutContainer.idl\
+	XLayoutFlow.idl\
+	XLayoutFlowContainer.idl\
+	XLayoutRoot.idl\
+	XLayoutUnit.idl
+.ENDIF # ENABLE_LAYOUT == TRUE
+
 # ------------------------------------------------------------------
 
 .INCLUDE :  target.mk
cvs diff: Diffing offapi/com/sun/star/awt/tree
cvs diff: Diffing offapi/com/sun/star/chart
cvs diff: Diffing offapi/com/sun/star/chart2
cvs diff: Diffing offapi/com/sun/star/chart2/data
cvs diff: Diffing offapi/com/sun/star/configuration
cvs diff: Diffing offapi/com/sun/star/configuration/backend
cvs diff: Diffing offapi/com/sun/star/configuration/backend/xml
cvs diff: Diffing offapi/com/sun/star/configuration/bootstrap
cvs diff: Diffing offapi/com/sun/star/datatransfer
cvs diff: Diffing offapi/com/sun/star/datatransfer/clipboard
cvs diff: Diffing offapi/com/sun/star/datatransfer/dnd
cvs diff: Diffing offapi/com/sun/star/deployment
cvs diff: Diffing offapi/com/sun/star/deployment/test
cvs diff: Diffing offapi/com/sun/star/deployment/ui
cvs diff: Diffing offapi/com/sun/star/document
cvs diff: Diffing offapi/com/sun/star/drawing
cvs diff: Diffing offapi/com/sun/star/drawing/framework
cvs diff: Diffing offapi/com/sun/star/embed
cvs diff: Diffing offapi/com/sun/star/form
cvs diff: Diffing offapi/com/sun/star/form/binding
cvs diff: Diffing offapi/com/sun/star/form/component
cvs diff: Diffing offapi/com/sun/star/form/control
cvs diff: Diffing offapi/com/sun/star/form/inspection
cvs diff: Diffing offapi/com/sun/star/form/runtime
cvs diff: Diffing offapi/com/sun/star/form/submission
cvs diff: Diffing offapi/com/sun/star/form/validation
cvs diff: Diffing offapi/com/sun/star/formula
cvs diff: Diffing offapi/com/sun/star/frame
cvs diff: Diffing offapi/com/sun/star/frame/status
cvs diff: Diffing offapi/com/sun/star/gallery
cvs diff: Diffing offapi/com/sun/star/geometry
cvs diff: Diffing offapi/com/sun/star/graphic
cvs diff: Diffing offapi/com/sun/star/i18n
cvs diff: Diffing offapi/com/sun/star/image
cvs diff: Diffing offapi/com/sun/star/inspection
cvs diff: Diffing offapi/com/sun/star/installation
cvs diff: Diffing offapi/com/sun/star/ldap
cvs diff: Diffing offapi/com/sun/star/linguistic2
cvs diff: Diffing offapi/com/sun/star/logging
cvs diff: Diffing offapi/com/sun/star/mail
cvs diff: Diffing offapi/com/sun/star/media
cvs diff: Diffing offapi/com/sun/star/mozilla
cvs diff: Diffing offapi/com/sun/star/packages
cvs diff: Diffing offapi/com/sun/star/packages/manifest
cvs diff: Diffing offapi/com/sun/star/packages/zip
cvs diff: Diffing offapi/com/sun/star/plugin
cvs diff: Diffing offapi/com/sun/star/presentation
cvs diff: Diffing offapi/com/sun/star/presentation/textfield
cvs diff: Diffing offapi/com/sun/star/rendering
cvs diff: Diffing offapi/com/sun/star/report
cvs diff: Diffing offapi/com/sun/star/report/inspection
cvs diff: Diffing offapi/com/sun/star/resource
cvs diff: Diffing offapi/com/sun/star/scanner
cvs diff: Diffing offapi/com/sun/star/script
cvs diff: Diffing offapi/com/sun/star/script/browse
cvs diff: Diffing offapi/com/sun/star/script/provider
cvs diff: Diffing offapi/com/sun/star/sdb
cvs diff: Diffing offapi/com/sun/star/sdb/application
cvs diff: Diffing offapi/com/sun/star/sdb/tools
cvs diff: Diffing offapi/com/sun/star/sdbc
cvs diff: Diffing offapi/com/sun/star/sdbcx
cvs diff: Diffing offapi/com/sun/star/security
cvs diff: Diffing offapi/com/sun/star/setup
cvs diff: Diffing offapi/com/sun/star/sheet
cvs diff: Diffing offapi/com/sun/star/smarttags
cvs diff: Diffing offapi/com/sun/star/style
cvs diff: Diffing offapi/com/sun/star/svg
cvs diff: Diffing offapi/com/sun/star/sync
cvs diff: Diffing offapi/com/sun/star/sync2
cvs diff: Diffing offapi/com/sun/star/system
cvs diff: Diffing offapi/com/sun/star/table
cvs diff: Diffing offapi/com/sun/star/task
cvs diff: Diffing offapi/com/sun/star/text
cvs diff: Diffing offapi/com/sun/star/text/fieldmaster
cvs diff: Diffing offapi/com/sun/star/text/textfield
cvs diff: Diffing offapi/com/sun/star/text/textfield/docinfo
cvs diff: Diffing offapi/com/sun/star/ucb
cvs diff: Diffing offapi/com/sun/star/ucb/smart
cvs diff: Diffing offapi/com/sun/star/ui
cvs diff: Diffing offapi/com/sun/star/ui/dialogs
cvs diff: Diffing offapi/com/sun/star/util
cvs diff: Diffing offapi/com/sun/star/view
cvs diff: Diffing offapi/com/sun/star/xforms
cvs diff: Diffing offapi/com/sun/star/xml
cvs diff: Diffing offapi/com/sun/star/xml/crypto
cvs diff: Diffing offapi/com/sun/star/xml/crypto/sax
cvs diff: Diffing offapi/com/sun/star/xml/csax
cvs diff: Diffing offapi/com/sun/star/xml/dom
cvs diff: Diffing offapi/com/sun/star/xml/dom/events
cvs diff: Diffing offapi/com/sun/star/xml/dom/views
cvs diff: Diffing offapi/com/sun/star/xml/input
cvs diff: Diffing offapi/com/sun/star/xml/sax
cvs diff: Diffing offapi/com/sun/star/xml/wrapper
cvs diff: Diffing offapi/com/sun/star/xml/xpath
cvs diff: Diffing offapi/com/sun/star/xsd
cvs diff: Diffing offapi/drafts
cvs diff: Diffing offapi/drafts/com
cvs diff: Diffing offapi/drafts/com/sun
cvs diff: Diffing offapi/drafts/com/sun/star
cvs diff: Diffing offapi/drafts/com/sun/star/form
cvs diff: Diffing offapi/prj
cvs diff: Diffing offapi/type_reference
cvs diff: Diffing offapi/util
